rootfs/etc/nginx/lua/monitor/logging.lua

```diff
@@ -0,0 +1,70 @@
+local statsd = require('statsd')
+local defer = require("util.defer")
+local split = require("util.split")
+
+local _M = {}
+
+local function send_response_data(upstream_state, client_state)
+  local status_class
+  if upstream_state.status then
+    for i, status in ipairs(upstream_state.status) do
+      -- TODO: link this with the zones when we use openresty-upstream
+      if status == '-' then
+        status = 'ngx_error'
+        status_class = 'ngx_error'
+      else
+        status_class = string.sub(status, 0, 1) .. "xx"
+      end
+      ngx.log(ngx.INFO, upstream_state.addr[i] )
+      statsd.increment('ingress_test.nginx.upstream.response', 1, {status=status,
+                                                      status_class=status_class,
+                                                      upstream_name=client_state.upstream_name})
+
+      statsd.histogram('ingress_test.nginx.upstream.response_time',
+                     upstream_state.response_time[i],
+                     {upstream_name=client_state.upstream_name})
+    end
+  end
+  status_class = string.sub(client_state.status, 0, 1) .. "xx"
+  statsd.increment('ingress_test.nginx.client.response', 1, {status=client_state.status,
+                                                status_class=status_class,
+                                                upstream_name=client_state.upstream_name,
+                                                scheme=client_state.scheme})
+
+  statsd.histogram('ingress_test.nginx.client.request_time', client_state.request_time, {
+    upstream_name=client_state.upstream_name
+  })
+end
+
+function _M.call()
+  local status, status_err = split.split_upstream_var(ngx.var.upstream_status)
```

Comment:
How does this work in the case of multiple upstream retries? It will have responses from all tried upstreams, is it gonna return the first one? Should it not return the last one rather?

Reply:
never mind, it returns all of them as a table then we emit them all to statsd